### PR TITLE
Fix for cyclonedds-cxx/idlcxx issue #70

### PR DESCRIPTION
**Pull request by reicheratwork**
Multiple deserialization is prevented by:

- the getT() calls no longer are preceded by deserialization of
the data, since that already happens in the getT call
- added a separate setT function which creates the correct sample
in-place without having to deserialize the data as with getT, allowing
skipping of one deserialization step

@sumanth-nirmal @e-hndrks : could you check whether this works as expected and that I did not forget anything?

### CODE REVIEW

**Review by sumanth-nirmal — CHANGES_REQUESTED** [review]

**Comment by reicheratwork** [comment]
Improved the implementation by splitting out a separate setter function `setT` which does not do unnecessary deserializations.

@sumanth-nirmal could you check this?

File: src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp [hunk 1]
```diff
@@ -74,6 +74,23 @@ class ddscxx_serdata : public ddsi_serdata {
...
+      t = new T(*toset);
+      T* exp = nullptr;
+      if (!m_t.compare_exchange_strong(exp, t, std::memory_order_seq_cst)) {
+        delete t;
+        t = exp;
+      }
+    } else {
+      *t = *toset;
+    }
+    return t;
```
Comment (sumanth-nirmal) [inline src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp:91]: We don't need to return anything, correct?

**Comment by reicheratwork** [comment]
@sumanth-nirmal The function could just as well return void, but as this function changes the contents of the entity, I think it is a good habit to return the thing that has changed (in this case the pointer of the contained sample).
Also the symmetry with the getT() function also returning the pointer to the sample is another reason why I thought that returning the sample pointer is a good idea.

**Comment by sumanth-nirmal** [comment]
@reicheratwork I would agree. The changes look good from me. @e-hndrks Can you probably also glance over?

**Review by sumanth-nirmal — APPROVED** [review]

**Comment by k0ekk0ek** [comment]
Included in #79 as well.

**Pull request merged**